lib/traveller.rb

```diff
@@ -32,7 +32,7 @@ def initialize(input)
   end
 
   def parse_zip
-    zip = @input[/\b([0-9]{5})\b/]
+    zip = @input[/\b([0-9]{5})|([abceghjklmnprstvxy]\d[abceghjklmnprstvwxyz]( )?\d[abceghjklmnprstvwxyz]\d)\b/]
```

Comment:
To help readability here we could separate the regex/create some helper methods:

```ruby
def parse_zip
  zip = parse_us_zip
  zip = parse_canadian_zip if zip.nil?
  @input.sub!(zip, '') if zip.present?
  zip
end

def parse_us_zip
  @input.match(US_ZIP_REGEX)
end

def parse_canadian_zip
  @input.match(CANADIAN_ZIP_REGEX)
end
```